mypyc/genops.py

```diff
@@ -713,7 +713,15 @@ def visit_assignment_stmt(self, stmt: AssignmentStmt) -> None:
             # name binding as a side effect.
             self.get_assignment_target(lvalue)
             return
-        self.assign(lvalue, stmt.rvalue)
+
+        line = stmt.rvalue.line
+        rvalue_reg = None
+        for lvalue in stmt.lvalues:
+            target = self.get_assignment_target(lvalue)
+            if rvalue_reg is None:
+                rvalue_reg = self.accept(stmt.rvalue)
```

Comment:
Some quick testing reveals that the RHS is evaluated before the LHS in assignments, so evaluating the RHS should happen before evaluating any LHSs.

Reply:
If that is what was causing the test churn you talked about earlier, then that's not just register ordering but a correctness bug, so we should just change it.